IRuettae/Core/IRuettae.Core.GeneticAlgorithm/Algorithm/Models/GenAlgStarterData.cs

```diff
@@ -26,9 +32,139 @@ public static GenAlgStarterData GetDefault(OptimizationInput input)
+                // 50 -> 131072
+                // 100 -> 16384
+                // 200 -> 16
+                // 1000 -> 16
+
+                if (x > 35 && x <= 50) // (35,50]
+                {
+                    // linear interpolation
+                    // generated with https://mycurvefit.com/
+                    // linear fit method: linear regression
+                    // y = -8738.133*x + 567978.7
+                    y = Math.Round(-8738.133 * x + 567978.7);
+                }
+                else if (x > 50 && x < 200) // (50,200)
+                {
+                    // non-linear approximation
+                    // generated with https://mycurvefit.com/
+                    // non-linear fit method: exponential basic
+                    // y = -250.92 + 1036717 * e ^ (-0.0413231 * x)
+                    y = Math.Round(-250.92 + 1036717 * Math.Pow(Math.E, (-0.0413231 * x)));
+                }
+                else if (x >= 200) // [200,inf]
+                {
+                    y = sizeBigger;
+                }
+                else // [-inf,35]
+                {
+                    y = sizeDefault;
+                }
+
+                starterData.PopulationSize = (int)y;
+            }
+            #endregion PopulationSize
+
             return starterData;
         }
 
+        /// <summary>
+        /// use GetDefault
+        /// </summary>
+        private GenAlgStarterData()
+        {
+
+        }
+
+        public GenAlgStarterData(int maxNumberOfSantas, long maxNumberOfGenerations, int populationSize, double elitismPercentage, double directMutationPercentage, double randomPercentage, double orderBasedCrossoverProbability, double mutationProbability, double positionMutationProbability)
+        {
+            MaxNumberOfSantas = maxNumberOfSantas;
+            MaxNumberOfGenerations = maxNumberOfGenerations;
+            PopulationSize = populationSize;
+            ElitismPercentage = elitismPercentage;
+            DirectMutationPercentage = directMutationPercentage;
+            RandomPercentage = randomPercentage;
+            OrderBasedCrossoverProbability = orderBasedCrossoverProbability;
+            MutationProbability = mutationProbability;
+            PositionMutationProbability = positionMutationProbability;
+        }
+
+        /// <summary>
+        /// Returns if this is a valid configuration.
+        /// </summary>
+        /// <returns></returns>
+        public bool IsValid()
+        {
+            if (MaxNumberOfSantas < 1)
+            {
+                return false;
+            }
+            if (MaxNumberOfGenerations < 0)
+            {
+                return false;
+            }
+            // one individual for elitism plus one "normal" individual
+            if (PopulationSize < 2)
+            {
+                return false;
+            }
+
+            // check percentages
+            if (ElitismPercentage <= 0 || ElitismPercentage >= 1)
```

Comment:
absichtliich hier <= und >= ?

Reply:
Ja.
Der Elitismus muss über 0 sein, damit die beste gefundene Lösung nicht verloren geht.
Der Elitismus muss kleiner als 1 sein, weil sich sonst die Population nicht verändern kann. Das heiss die Population wird auch nach x Generationen noch dieselbe sein, da immer die gesamt Population wegen dem Elitismus in die nächste Generation übernommen wird.

Reply:
Es wird übrigens bei der Evolutions sichergestellt, dass mindestens das beste Individuum in die nächste Generation übernommen wird. D.h. auch bei einem Elitismus von <=0 würde es auch funktionieren.
Jedoch hatte ich das Gefühl, dass dies bei der Validation schon klargemacht werden sollte.